src/components/Subplaylist.tsx

```diff
@@ -42,19 +50,82 @@ const useStyles = makeStyles(theme => ({
     height: 230,
     overflow: 'auto'
   },
+  wrapper: {
+    margin: theme.spacing(1),
+    position: 'relative',
+  },
   button: {
-    margin: theme.spacing(0.5, 0)
-  }
+    margin: theme.spacing(0.5, 0),
+    size: 'medium'
+  },
+  buttonSuccess: {
+    backgroundColor: green[500],
+    '&:hover': {
+      backgroundColor: green[700],
+    },
+  },
+  buttonProgress: {
+    color: green[500],
+    position: 'absolute',
+    top: '50%',
+    left: '40%',
+    marginTop: -12,
+    marginLeft: -8,
+  },
 }))
 
+export interface SnackbarMessage {
+  message: string;
+  key: number;
+}
+
+export interface State {
+  open: boolean;
+  snackPack: SnackbarMessage[];
+  messageInfo?: SnackbarMessage;
+}
+
+
 export default function Subplaylist (props: {
   source: PlaylistObj
   playlist: PlaylistObj
   genres: string[]
   onDelete?: (playlist: PlaylistObj) => any
 }) {
+
   const classes = useStyles();
 
+  // SAVING ANIMATION STUFF
+  const [loading, setLoading] = React.useState(false);
+  const [success, setSuccess] = React.useState(false);
+  const [buttonLabel, setButtonLabel] = useState("Save")
+
+  const buttonClassname = clsx({
+    [classes.button]: true,
+    [classes.buttonSuccess]: success,
+  });
+
+  const handleButtonClick = async () => {
+    if (!loading) {
+      // Start loading animation while saving playlist
+      setSuccess(false);
+      setLoading(true);
+      const user = await getUserProfile();
+      await createOrUpdatePlaylist(user.id, props.playlist);
+      // wait 2 seconds before telling the user the playlist has saved
+      setTimeout(() => {
+        setButtonLabel("Saved");
+        setSuccess(true);
+        setLoading(false);
+      },2000);
```

Comment:
_dis kinda dodg ya feel?_

I'd think that the time it takes for both of the API calls should already be an adequate delay.

Reply:
LOL yeah I tried without the timeouts at first, but the api calls happen pretty quick. The timeouts don't block anything else so should be fine?

Reply:
When your internet is too fast :')

Perhaps some sort of time comparison to have it as a minimum of `X` seconds, rather than `delay + X`.
But yeah, PR lgtm